crates/revm/src/evm_impl.rs

```diff
@@ -111,22 +111,22 @@ impl<'a, GSPEC: Spec, DB: Database, const INSPECT: bool> Transact<DB::Error>
             .info
             .balance;
 
-        let payment_value = U256::from(gas_limit)
-            .checked_mul(effective_gas_price)
+        let balance_check = U256::from(gas_limit)
+            .checked_mul(self.data.env.tx.gas_price)
             .ok_or(EVMError::Transaction(
                 InvalidTransaction::OverflowPaymentInTransaction,
             ))?;
 
         // Check if account has enough balance for gas_limit*gas_price and value transfer.
         // Transfer will be done inside `*_inner` functions.
-        if payment_value + value > *caller_balance && !disable_balance_check {
+        if balance_check + value > *caller_balance && !disable_balance_check {
```

Comment:
I know it was not present before, but this too needs checked/safe math as it can overflow and panic, can you add it?

Reply:
done